### PR TITLE
#7 added mock to answerers

### PR DESCRIPTION
**Pull request by g-harel**

### CODE REVIEW

File: searx/answerers/__init__.py [hunk 1]
```diff
@@ -42,10 +42,10 @@ def ask(self, query):
         results = []
         query_parts = list(filter(None, query.query.split()))
 
-        if query_parts[0].decode('utf-8') not in answerers_by_keywords:
+        if query_parts[0] not in answerers_by_keywords:
```
Comment (Radu-Raicea) [inline searx/answerers/__init__.py]: You should probably learn the basics of encoding and their differences between Python 2 and 3.

**Comment by Radu-Raicea** [comment]
Can you please reference the issue number in the last commit? @g-harel

File: searx/answerers/__init__.py [hunk 2]
```diff
@@ -23,28 +21,32 @@ def load_answerers():
...
 
-def get_answerers_by_keywords(answerers):
-    by_keyword = defaultdict(list)
-    for answerer in answerers:
-        for keyword in answerer.keywords:
-            for keyword in answerer.keywords:
-                by_keyword[keyword].append(answerer.answer)
-    return by_keyword
-
+class Answerers():
```
Comment (Radu-Raicea) [inline searx/answerers/__init__.py:24]: Do we need to create a class for this? Shawn’s mocks assuming it’s just a function.

Comment (g-harel) [reply searx/answerers/__init__.py:24]: I'm gonna assume you meant "Shawn's mocks are just a function" ?

Comment (Radu-Raicea) [reply searx/answerers/__init__.py:24]: Shawn’s mocks assume it’s just a function.*

Comment (g-harel) [reply searx/answerers/__init__.py:24]: well I'm making the loader pluggable and it is used by all three functions. I didn't want to keep state in the file/module as a variable

**Comment by g-harel** [comment]
@Radu-Raicea I'm gonna be squashing

File: searx/answerers/__init__.py (hunk 2 above)
Comment (g-harel) [reply searx/answerers/__init__.py:24]: Looking at his commit, it seems he will just need to change the import and the line it's used at.

```python
from searx.answerers import Answerers, file_loader
# ...
Answerers(file_loader).ask(...)
```

vs.

```python
from searx.answerers import ask
# ...
ask(...)
```

File: searx/answerers/__init__.py (hunk 1 above)
Comment (g-harel) [reply searx/answerers/__init__.py]: enlighten me

**Review by emd21 — APPROVED** [review]

**Pull request merged**